src/mpi/coll/ialltoallv/ialltoallv_tsp_scattered_algos.h

```diff
@@ -105,11 +104,11 @@ int MPIR_TSP_Ialltoallv_sched_intra_scattered(const void *sendbuf, const int sen
         for (j = 0; j < ww; j++) {
             src = (rank + i + j) % size;
             recv_id[(i + j) % bblock] =
-                MPIR_TSP_sched_irecv((char *) recvbuf + rdispls[src],
+                MPIR_TSP_sched_irecv((char *) recvbuf + rdispls[src] * recvtype_extent,
```

Comment:
This is the real bug. I was staring at it for hours and failed to spot the obvious. This is a high example that having lines span multiple lines hinders the reading.